superset/db_engine_specs/sqlite.py

```diff
@@ -82,7 +82,7 @@ def convert_dttm(
             utils.TemporalType.DATETIME,
             utils.TemporalType.TIMESTAMP,
         ):
-            return f"""'{dttm.isoformat(sep=" ", timespec="seconds")}'"""
+            return f"""'{dttm.isoformat(sep=" ", timespec="microseconds")}'"""
```

Comment:
I remember that this line was fixed by [PR](https://github.com/apache/superset/pull/21113) because the SQLite manual said that _The datetime() function returns the date and time as text in their same formats: YYYY-MM-DD HH:MM:SS._([link](https://www.sqlite.org/lang_datefunc.html))

So, please check out the previous PR and prevent introducing a regression. Thanks.